### PR TITLE
rgw: radogw-admin reshard status command should print text for reshard status

### PR DESCRIPTION
**Pull request by oritwas**
Fixes: http://tracker.ceph.com/issues/23257
Signed-off-by: Orit Wasserman <owasserm@redhat.com>

### CODE REVIEW

File: src/cls/rgw/cls_rgw_types.cc [hunk 1]
```diff
@@ -616,7 +616,21 @@ void cls_rgw_reshard_entry::generate_test_instances(list<cls_rgw_reshard_entry*>
...
 void cls_rgw_bucket_instance_entry::dump(Formatter *f) const
 {
-  encode_json("reshard_status", (int)reshard_status, f);
+  string status_str;
+  switch(reshard_status) {
+    case CLS_RGW_RESHARD_NONE:
+      status_str= "none";
+      break;
+    case CLS_RGW_RESHARD_IN_PROGRESS:
+      status_str = "in progress";
```
Comment (yehudasa) [inline src/cls/rgw/cls_rgw_types.cc]: maybe `in-progress` instead?

File: src/rgw/rgw_admin.cc [hunk 2]
```diff
@@ -5834,7 +5834,11 @@ int main(int argc, const char **argv)
       return -r;
     }
 
-    encode_json("status", status, formatter);
+    for (auto s: status) {
+      formatter->open_object_section("status");
+      s.dump(formatter);
+      formatter->close_section();
```
Comment (yehudasa) [inline src/rgw/rgw_admin.cc]: `encode_json("status", s, formatter);` instead?

**Comment by oritwas** [comment]
@yehudasa ping

File: src/rgw/rgw_admin.cc [hunk 3]
```diff
@@ -5834,7 +5834,9 @@ int main(int argc, const char **argv)
       return -r;
     }
 
-    encode_json("status", status, formatter);
+    for (auto s: status) {
+      encode_json("status", s, formatter);
```
Comment (yehudasa) [inline src/rgw/rgw_admin.cc]: @oritwas how does the output here look like? Should we open a formatter array section?

**Comment by oritwas** [comment]
@yehudasa  changes to rgw_admin.cc were leftovers I removed them

**Comment by yuriw** [comment]
@yehudasa pls rebase

--- pr 20779 --- pulling https://github.com/oritwas/ceph.git branch wip-rgw-23257
remote: Counting objects: 6, done.
remote: Total 6 (delta 5), reused 5 (delta 5), pack-reused 1
Unpacking objects: 100% (6/6), done.
From https://github.com/oritwas/ceph
 * branch            wip-rgw-23257 -> FETCH_HEAD
CONFLICT (file/directory): There is a directory with name src/dmclock in 6be9906890be24eb293b016e5638ef84848d690d. Adding src/dmclock as src/dmclock~HEAD
Auto-merging src/cls/rgw/cls_rgw_types.cc
Adding qa/suites/rbd/mirror/cluster
Adding qa/suites/rbd/mirror/base
Adding qa/suites/ceph-deploy/basic/distros/ubuntu_latest.yaml
Adding qa/suites/ceph-deploy/basic/distros/centos_latest.yaml
Adding qa/suites/ceph-deploy/basic/distros/.qa
Automatic merge failed; fix conflicts and then commit the result.
Traceback (most recent call last):
  File "/home/yuriw/wip_master/src/script/build-integration-branch", line 62, in <module>
    assert not r
AssertionError

**Comment by oritwas** [comment]
@yuriw I rebased to upstream master.
This pr doesn't have anything related to dmclock submoudle, it could be a left over from a different pr you are testing.

**Comment by yuriw** [comment]
[wip-yuri2-testing-2018-07-05-2019](https://trello.com/c/xy0ckEBF/337-wip-yuri2-testing-2018-07-05-2019)

**Pull request merged**